### PR TITLE
chore(unit tests): Adding additional unit tests

### PR DESCRIPTION
**Pull request by ruisebas**
## Description
This PR adds additional unit tests to the following targets:
- AWSCognitoAuthPlugin
- AWSStoragePlugin
- AWSPinpointPushNotificationsPlugin

It also removes some orphaned code found around :)

## General Checklist

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSAuthCognitoSession.swift [hunk 1]
```diff
@@ -76,24 +76,6 @@ public struct AWSAuthCognitoSession: AuthSession,
 
 }
 
-/// Internal Helpers for managing session tokens
-internal extension AWSAuthCognitoSession {
-    func areTokensExpiring(in seconds: TimeInterval? = nil) -> Bool {
```
[the remark this answers is not in the document]
Comment (ruisebas) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSAuthCognitoSession.swift:81]: I hope not 😅 
We can always bring it back if we do

File: AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageTransferTask.swift [hunk 2]
```diff
@@ -324,7 +320,7 @@ class StorageTransferTask {
                 logger.warn("Unable to complete after cancelled")
                 return
             }
-            guard _status == .completed else {
+            guard _status != .completed else {
```
[the remark this answers is not in the document]
Comment (ruisebas) [reply AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageTransferTask.swift:323]: I still don't understand how it's working right now 😆

File: AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/CognitoASFTests/CognitoUserPoolASFTests.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+@testable import AWSCognitoAuthPlugin
+import XCTest
+
+class CognitoUserPoolASFTests: XCTestCase {
+    private var pool: CognitoUserPoolASF!
```
Comment (thisisabhash) [inline AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/CognitoASFTests/CognitoUserPoolASFTests.swift:12]: nit: rename to `userPool`?

File: AmplifyPlugins/Storage/Tests/AWSS3StoragePluginTests/Dependency/AWSS3AdapterTests.swift [hunk 4]
```diff
@@ -0,0 +1,755 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+@testable import Amplify
+//@testable import func AmplifyTestCommon.XCTAssertThrowFatalError
```
Comment (thisisabhash) [inline AmplifyPlugins/Storage/Tests/AWSS3StoragePluginTests/Dependency/AWSS3AdapterTests.swift:9]: can remove the comment if not needed

**Review by thisisabhash — APPROVED** [review]
LGTM, minor comments

**Pull request merged**